src/bank02.asm

```diff
@@ -8130,29 +8126,197 @@ titleScreenLicenseText:
+.divmod_and_store_number:
+    ld   A, $0a
+    push BC
+    push DE
+    call divMod
+    pop  DE
+    pop  BC
+    push AF
+    inc  B
+    ld   A, H
+    or   A, L
+    jr   NZ, .divmod_and_store_number
+
+    ; Reduce second nibble of C by num digits found
+    ; Remainder provides the number of tiles to clear
+    ; after we write the digits
+    ld   A, C
+    sub  A, B
+    ld   C, A
+
+    ; Increment tile buffer pointer (DE) if in swap mode
+    ; since we will be swapping back 4px to the tile before
+    ld   H, D
+    ld   L, E
+    ld   D, $00
+    and  A, $10
+    ld   E, A
+    add  HL, DE
+    ld   D, H
+    ld   E, L
+
+    ; Stack holds B digits to pop, transfer them
+ .grab_digit_and_transfer:
+    ld   HL, gfxStatusBar+$100 ; start of number tiles
+    pop  AF ; grab digit
+    swap A
+    push DE
+    ld   D, $00
+    ld   E, A
+    add  HL, DE
+    pop  DE
+    ld   A, BANK(gfxStatusBar)
+    push BC
+    ld   B, $10
+    call copyAndRotateBankAfromHLtoDE
+    pop  BC
+    dec  B
+    jr   NZ, .grab_digit_and_transfer
+
+    ; Clear remaining tiles
+    ld   A, C
+    and  A, $0f
+    jr   Z, .check_swap
+
+    ld   H, D
+    ld   L, E
+    ld   B, A
+    xor  A, A
+ .clear_remaining_tiles:
+    call memsetTileWithA
+    dec  B
+    jr   NZ, .clear_remaining_tiles
+
+    ; Shifted tiles are in WRAM, check if swap needed
+ .check_swap:
+    pop  DE
+    ld   L, C
+    pop  BC
+    bit  4, C
+    ret  Z
+
+    ; We need to swap back tiles, prepare...
+    push BC
+    push DE
+    ld   A, C
+    sub  A, L
+    ld   B, A ; B holds number of tiles to swap
+    call swapBackHalfTiles
+    pop  DE
+    pop  BC
+    ret
+
+; Swaps tiles back across half tile boundaries
+; B holds number of tiles to shift
+; DE holds address of first tile
+swapBackHalfTiles:
+    swap B ; now B holds number of bytes to swap
+
+    ; Clear the first tile (it was left untouched so far)
+    ld   H, D
+    ld   L, E
+    xor  A, A
+    call memsetTileWithA
+
+    ; Swap HL and DE
+    ld   C, H
+    ld   H, D
+    ld   D, C
+    ld   C, L
+    ld   L, E
+    ld   E, C
+
+.shift_4px:
+    ; First "or" in the first 4 bits of the next tile byte
+    ; into the current tile byte
+    ld   A, [DE]
+    swap A
+    ld   C, A
+    and  A, $0f
+    or   A, [HL]
+    ld   [HL+], A
+    ; Now shift up the last 4 bits of the next tile byte
+    ld   A, C
+    and  A, $f0
+    ld   [DE], A
+    inc  DE
+    dec  B
+    jr   NZ, .shift_4px
+    ret
+
+; Requests transfer of status bar WRAM tiles to VRAM
+; B holds VRAM tile position
+; C holds the drawing mode in the first nibble (0 for shift, 1 for shift/swap)
+; C holds number of max digits in the second nibble
+; DE holds the WRAM address
+requestVRAMStatusBarTransfer:
+    ; Short circuit if nothing to do, else load number of tiles in C
+    ; This logic adds one to max digits if mode is shift-swap
+    ld   A, C
+    and  A, $10
+    swap A
+    add  A, C
+    and  A, $0f
+    ret  Z
+    ld   C, A
```

Comment:
How about:
`bit  4, C`
`jr   Z, .test`
`inc  C`
`and  A, $0f`
`ret  Z`

It's one byte shorter, but is it as clear or clearer?

Reply:
A isn't loaded yet, so the `and A, $0f` doesn't work.